plone/app/querystring/querybuilder.py

```diff
@@ -18,10 +16,56 @@
 
 import json
 import logging
+import re
+
 
 logger = logging.getLogger("plone.app.querystring")
 _ = MessageFactory("plone")
 
+# We should accept both a simple space, unicode u'\u0020 but also a
+# multi-space, so called 'waji-kankaku', unicode u'\u3000'
+_MULTISPACE = '\u3000'
+_BAD_CHARS = ('?', '-', '+', '*', _MULTISPACE)
+
+def _quote_chars(s):
+    # We need to quote parentheses when searching text indices
+    if '(' in s:
+        s = s.replace('(', '"("')
+    if ')' in s:
+        s = s.replace(')', '")"')
+    if _MULTISPACE in s:
+        s = s.replace(_MULTISPACE, ' ')
+    return s
+
+
+def _quote(term):
+    # The terms and, or and not must be wrapped in quotes to avoid
+    # being parsed as logical query atoms.
+    if term.lower() in ('and', 'or', 'not'):
+        term = '"%s"' % term
+    return term
+
+def munge_search_term(query):
```

Comment:
A quick search shows that it is only used here, on `CMFPlone/browser/search.py` and its related tests. Do you plan to make CMFPlone import this and move also the tests here?

Should this rather not go into `plone.base` instead, though? I see potentially other distributions (collective.solr, collections (`plone.app.content` or `contenttypes`, `plone.app.portlets` ?) that might reuse this...

As a first step to untangle the dependency is fine, but maybe we can do this move to `plone.base`?

Reply:
it is used here and in CMFPlone. So I do not expect it to be used somewhere else. I would keep it here and later on move tests here and import in CMFPlone frim here.

Reply:
If this is OK I will create a ticket in CMFPlone so we don't forget that